sw/device/silicon_creator/lib/ownership/ownership.c

```diff
@@ -227,3 +274,40 @@ rom_error_t ownership_flash_lockdown(boot_data_t *bootdata,
   }
   return kErrorOk;
 }
+
+void ownership_pages_lockdown(boot_data_t *bootdata, hardened_bool_t rescue) {
+  flash_ctrl_perms_t perm = {
+      .read = kMultiBitBool4True,
+      .write = kMultiBitBool4False,
+      .erase = kMultiBitBool4False,
+  };
+  flash_ctrl_cfg_t cfg = {
+      .scrambling = kMultiBitBool4True,
+      .ecc = kMultiBitBool4True,
+      .he = kMultiBitBool4False,
+  };
+  // Always make page 0 read only.
+  flash_ctrl_info_perms_set(&kFlashCtrlInfoPageOwnerSlot0, perm);
+  flash_ctrl_info_cfg_set(&kFlashCtrlInfoPageOwnerSlot0, cfg);
+  abs_mmio_write32(kFlashCtrlInfoPageOwnerSlot0.cfg_wen_addr, 0);
```

Comment:
It seems `flash_ctrl_info_page_t` needs to be updated to add `cfg_wen_addr`.

Reply:
It seems like that `cfg_wen_addr` got renamed to `cfg_wen_offset` and is to be added to a `base_addr`.